src/Umbraco.Infrastructure/Persistence/Repositories/Implement/UserRepository.cs

```diff
@@ -1070,5 +1070,45 @@ public IEnumerable<IUser> GetNextUsers(int id, int count)
             : GetMany(ids).OrderBy(x => x.Id) ?? Enumerable.Empty<IUser>();
     }
 
+    /// <inheritdoc />
+    public void InvalidateSessionsForRemovedProviders(IEnumerable<string> currentProviderKeys)
+    {
+        // Get all the user or member keys associated with the removed providers.
+        Sql<ISqlContext> idsQuery = SqlContext.Sql()
+            .Select<ExternalLoginDto>(x => x.UserOrMemberKey)
+            .From<ExternalLoginDto>()
+            .WhereNotIn<ExternalLoginDto>(x => x.ProviderKey, currentProviderKeys);
+        List<Guid> userAndMemberKeysAssociatedWithRemovedProviders = Database.Fetch<Guid>(idsQuery);
+
```

Comment:
Should probably exit early if there is no work to be done 😄 

```suggestion
        if (userAndMemberKeysAssociatedWithRemovedProviders.Count == 0)
        {
            return;
        }

```